detector/test_helper.go

```diff
@@ -66,14 +68,29 @@ func TestDetectByCreatorName(creatorMethod string, src string, stateJSON string,
 	if preprocess := detector.MethodByName("PreProcess"); preprocess.IsValid() {
 		preprocess.Call([]reflect.Value{})
 	}
-	for _, template := range schema {
-		for _, resource := range template.Find("resource", reflect.Indirect(detector).FieldByName("Target").String()) {
-			detect := detector.MethodByName("Detect")
-			detect.Call([]reflect.Value{
-				reflect.ValueOf(resource),
-				reflect.ValueOf(issues),
-			})
+	switch reflect.Indirect(detector).FieldByName("TargetType").String() {
+	case "resource":
```

Comment:
__[gometalinter]__
* [goconst] 2 other occurrence(s) of "resource" found in: detector.go:203:7 detector.go:278:7

<!-- Comment by [SideCI](https://sideci.com) -->